hypex/dataset/backends/abstract.py

```diff
@@ -195,11 +195,19 @@ def _get_column_index(
         raise AbstractMethodError
 
     @abstractmethod
-    def get_column_type(self, column_name: str) -> str:
+    def get_column_type(self, column_name: str) -> Union[DefaultRoleTypes, None]:
+        raise AbstractMethodError
+    
+    @abstractmethod
+    def astype(
+        self,
+        dtype: Dict[str, DefaultRoleTypes],
+        errors: Literal["raise", "ignore"] = "raise"
+    ) -> Any:
         raise AbstractMethodError
 
     @abstractmethod
-    def update_column_type(self, column_name: str, type_name: str):
+    def update_column_type(self, column_name: str, type_name: DefaultRoleTypes):
```

Comment:
type name should be passed as str or type object, not as an object of one of the DefaultRoleTypes